bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -558,18 +604,35 @@ public HttpsURLConnection makeRequest(String verb, String url, @Nullable String
                         }
                     }
                 }
+                if (code == 400 && repeatBadRequest) {
+                    badRequestCounter++;
+                    if (badRequestCounter < 3) {
+                        try {
+                            Thread.sleep(500);
```

Comment:
This seems really too blocking, can you please look into solving this by scheduling a new check instead of sleeping?

Sleeps do not belong in production code, they keep threads waiting and will cause problems in bigger openHAB instances.

Reply:
This is a very complicated uncommon case. The reason is, the servers seams to have an input queue with a limited size. This causes a bad request result if the binding makes to many calls in a too short timespan. In my installation does this never occur, but it is reported by users who setting channels of more than 10 devices at the same time in a rule. But handling this in a central place is only possible here in this synchronous function. At this place, I have no possibility to use a timer. The only other possibility would be a handling in the general command handler, but in this case, I cause more cpu load, because all the data preparation must be done again.

Reply:
@openhab/2-x-add-ons-maintainers what do you think about this.

Reply:
I find now a solution with a scheduler.